### DISCUSSION TITLE
Intermittent chunk load errors in Firefox

### DISCUSSION THREAD

**Discussion by karptonite** [webpack/webpack#15422] [Q&A]
# Bug report

**What is the current behavior?**
Intermittently, Firefox users of our Angular site (with lazy loading of chunks) will get ChunkLoadErrors that prevent our site from loading properly. This has been maddeningly intermittent and I have not been able to reproduce it, but I've collected some HAR files from users with details that may point in the right direction. This may be an issue with Firefox itself, or possibly plugins/add-ons/extension (although from the list of plugins users have reported, there is no plugin in common between them all, I believe, and some have said they have seen the issue when they run vanilla Firefox.)

I'm not super adept at decoding HAR files, but it is interesting is that when chunks fail to load, there appears to be something odd with the Protocol.
First, a chunk that loaded properly (it seems):
<img width="1507" alt="Screen Shot 2022-02-20 at 9 38 10 AM" src="https://user-images.githubusercontent.com/132278/154847979-dc7b65f4-bc23-4cea-aa2f-db6519526e85.png">

Now, from the same HAR, a chunk that did not load properly (notice status code 0, -1 B transferred). Notice also that the GET line in the raw Request omits HTTP/2. I believe that this file was the missing chunk that broke the page:
<img width="1512" alt="Screen Shot 2022-02-20 at 9 39 36 AM" src="https://user-images.githubusercontent.com/132278/154848043-126b2ff3-e333-4e84-bb28-046ab9b0f1cb.png">

 But also from the same HAR, there is this request, that seems to have loaded properly, but for some reason has status code 0? This one could just be loading from Cache or something, but I'm not sure.
<img width="1519" alt="Screen Shot 2022-02-20 at 9 41 56 AM" src="https://user-images.githubusercontent.com/132278/154848147-6c3faa06-d067-427f-8bba-0b812892024e.png">

In all cases, it seem that users are able to fix the problem by clearing their caches and reloading the page. From the HAR files, I've seen that the filenames after reload are the same as those that failed to load previously, so we aren't dealing new code being deployed resulting in missing old chunks.

For a bit more background with some additional HAR files, see [this StackOverflow question](https://stackoverflow.com/questions/71070943/what-is-the-source-of-this-intermittent-webpack-chunkloaderror-in-firefox). This has also been discussed at length on our site's forums, [here](https://boardgamegeek.com/thread/2788693/specific-videos-page-immediately-redirects-me-empt) and [here](https://boardgamegeek.com/thread/2800974/cant-open-forum-threads-firefox).

**If the current behavior is a bug, please provide the steps to reproduce.**
I suppose, load one of the threads linked to our site above and cross your fingers, but I've never been able to reproduce the issue myself.

**What is the expected behavior?**
Chunks should all load.

**Other relevant information:**
webpack version:  5.67.0
Node.js version: 14.18.1
Operating System: 
Additional tools: Angular-CLI

I have quite a few more HAR files, but I want to be careful how I share the data from them, because they contain some sensitive information.

**Comment by vankop** [comment]
it looks like it is some network problems, not sure how this related to webpack..

**Comment by karptonite** [comment]
It might be unrelated to webpack--I'm not sure what determines how the chunks are loaded. If it were just a network error, though, I would assume that clearing cache would not fix it--that is, I wouldn't expect a bad result to go into cache as a result of a network error.

**Reply by vankop** [reply]
it looks like there is no tls connection ( maybe bad network ). in protocol column there is no protocol, this means http request did not happened

**Reply by karptonite** [reply]
Right, but in the same HAR file, many requests did happen. And the problem persisted (for that one chunk only, I think) until the user cleared cache. If it were just a bad network, that is not what I'd expect. And, of course, there is the issue that this problem has been reported by at least half a dozen Firefox users (and only Firefox). If it were just the result of a bad network, I would expect it to affect all browsers.

**Reply by vankop** [reply]
sorry, there is no webpack related thing.. webpack uses jsonp (insert script tag on the page) to load chunks, so nothing to do from webpack side

**Comment by orrgottlieb** [comment]
Hi, we have the same issue, especially with Firefox, the amount of these errors that are happing on Firefox is extremely high in non proportional way to our Firefox users (nor to the amount of Firefox users world wide)

**Reply by alexander-akait** [reply]
Maybe timout in network

**Reply by orrgottlieb** [reply]
@alexander-akait , thanks, from what we are seeing we don't have any network issues, we will add additional information and will update

**Comment by roniavi** [comment]
Bump

**Reply by alexander-akait** [reply]
How you want to solve it? You can override webpack runtime API and implement own logic. Do you want it?

**Reply by hipstersmoothie** [reply]
Does webpack try to reload chunks when there are failures?

**Reply by alexander-akait** [reply]
No, are you looking for https://github.com/mattlewis92/webpack-retry-chunk-load-plugin?

**Reply by hipstersmoothie** [reply]
I'm a little surprised webpack doesn't do that by default. Thanks though! Ended up including that plugin
